app/src/test/java/com/codesoom/assignment/controllers/TaskControllerMock_MVC_Test.java

```diff
@@ -0,0 +1,173 @@
+    private static final String UPDATE_TITLE = "otherTest";
+    
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private TaskService taskService;
+
+    @Nested
+    @DisplayName("CREATE")
+    class Describe_create {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        @Test
+        @DisplayName("create메소드는 클라이언트가 요청한 새로운 Task를 Tasks에 추가해줍니다.")
+        void create() throws Exception {
+            Task task = new Task();
+            task.setTitle(TASK_TITLE_ONE);
+
+            given(taskService.createTask(any())).willReturn(task);
+
+            mockMvc.perform(post("/tasks")
+                            .content(objectMapper.writeValueAsString(task))
+                            .contentType(MediaType.APPLICATION_JSON))
+                            .andExpect(status().isCreated())
+                            .andExpect(content().string(containsString(TASK_TITLE_ONE)))
+                            .andDo(print());
+        }
+    }
+
+    @Nested
+    @DisplayName("READ")
+    class Describe_read {
+        @BeforeEach
+        void setUp() {
+            List<Task> tasks = new ArrayList<>();
+
+            Task task = new Task();
+            task.setTitle(TASK_TITLE_ONE);
+            tasks.add(task);
+
+            given(taskService.getTasks()).willReturn(tasks);
+
+            given(taskService.getTask(1L)).willReturn(task);
+        }
+
+        @Nested
+        @DisplayName("list 메소드는")
+        class Describe_list {
+            @Test
+            @DisplayName("Tasks에 있는 모든 Task를 반환합니다.")
+            void list() throws Exception {
+                mockMvc.perform(get("/tasks"))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string(containsString(TASK_TITLE_ONE)));
+            }
+        }
+
+        @Nested
+        @DisplayName("detail 메소드는")
+        class Describe_detail {
+            @Nested
+            @DisplayName("클라이언트가 요청한 Task의 id가 Tasks에 있으면")
+            class Context_with_valid_id {
+                @Test
+                @DisplayName("id에 해당하는 Task를 반환합니다.")
+                void detailWithValidId() throws Exception {
+                    mockMvc.perform(get("/tasks/1"))
+                            .andExpect(status().isOk())
+                            .andExpect(content().string(containsString(TASK_TITLE_ONE)));
+                }
+            }
+
+            @Nested
+            @DisplayName("클라이언트가 요청한 Task의 id가 Tasks에 없으면")
+            class Context_with_invalid_id {
+                @Test
+                @DisplayName("TaskNotFoundException 예외를 던집니다.")
+                void detailWithInvalidId() throws Exception {
+                    given(taskService.getTask(100L)).willThrow(new TaskNotFoundException(100L));
+
+                    mockMvc.perform(get("/tasks/100"))
+                            .andExpect(status().isNotFound());
+                }
+            }
```

Comment:
Context에서 말하고 있는 "상황"을 Context에서 직접 준비하도록 작업해 보세요.
```suggestion
            @Nested
            @DisplayName("클라이언트가 요청한 Task의 id가 Tasks에 없으면")
            class Context_with_invalid_id {

                final long requestTaskId = 100L;  // <---

                @Test
                @DisplayName("TaskNotFoundException 예외를 던집니다.")
                void detailWithInvalidId() throws Exception {
                    given(taskService.getTask(requestTaskId)).willThrow(new TaskNotFoundException(requestTaskId));

                    mockMvc.perform(get("/tasks/" + requestTaskId))
                            .andExpect(status().isNotFound());
                }
            }
```

Reply:
자세한 설명 감사합니다^^